### PR TITLE
OE-853/epub-to-webpub

### PR DESCRIPTION
**Pull request by kristojorg**
This PR brings the epub-to-webpub code into this monorepo. It also brings the git history from epub-to-webpub in as well by merging the branch from that remote directly into this repository. Not super important, but it can be helpful.

We are keeping:
- A vercel deployment with an API in use by DRB (I will configure this change once this is merged)
- A CLI built by [tsup](https://tsup.egoist.dev/#typescript--javascript) (a user friendly wrapper around esbuild) that is what we used before and what is recommended by turborepo. We can change to a different build system when the organization coalesces around one. I didn't think it was time to switch to Vite in this PR. 

**Note that the commits in this PR include all the commits from epub-to-webpub in order to bring that history into this repo**

### CODE REVIEW

**Review by Toxiapo — COMMENTED** [review]
All Looks good and works for me running locally. Just one small question on the react dependency

**Comment by samanthaandrews** [comment]
> Note that the commits in this PR include all the commits from epub-to-webpub in order to bring that history into this repo

This is cool!

I think we also need to do we need to do some sort of process around archiving the `epub-to-webpub` repo. Is that correct?  I've posted this as a question in Stack Overflow because I know there is an official process for this but I can't remember where those docs live: https://stackoverflowteams.com/c/nypl-digital/questions/236

I'll let you know when I get a response.

**Comment by samanthaandrews** [comment]
Ok this is the process for decommissioning and deprecating apps: https://github.com/NYPL/engineering-general/blob/main/standards/decommissioning.md

File: packages/epub-to-webpub/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "extends": "../../tsconfig.json",
+  "compilerOptions": {
+    "target": "ESNext",
+    "module": "ESNext",
+    "moduleResolution": "Node",
+    "jsx": "react"
+  },
+  "include": ["*.ts", "*.tsx", "chakra/**/*"]
```
[the remark this answers is not in the document]
Comment (kristojorg) [reply packages/epub-to-webpub/tsconfig.json:9]: Good question. This was probably just copied over from some other project at some point and never fully edited. I'll remove

**Comment by kristojorg** [comment]
Okay I archived epub-to-webpub and linked to this repo. I will also send a message in #dev

**Comment by samanthaandrews** [comment]
> Okay I archived epub-to-webpub and linked to this repo. I will also send a message in #dev

Cool thanks!

**Review by samanthaandrews — APPROVED** [review]

**Review by Toxiapo — APPROVED** [review]

**Comment by kristojorg** [comment]
@samanthaandrews I'm still getting some failing Cypress tests :/ arghhhh why does this stuff have to be so hard

**Comment by samanthaandrews** [comment]
@kristojorg I will look into these Cypress failures

**Comment by samanthaandrews** [comment]
I fixed the Cypress errors by re-running these tests 😅 The failures we were seeing were both on the tests that required us to open a book. We have been experiencing intermittent Axis360 timeout issues over the past week and I'm assuming these errors are related to that. We are mocking our fulfillment endpoint, but I don't think we're doing enough to bypass Axis360 entirely. Is it possible to do so?

This morning I sent an email to B&T to make sure they are aware of these issues. This is the Slack thread where Barbara posted about support emails that had come in overnight: https://nypl.slack.com/archives/C0153AFA1RS/p1677765866701909

I just did more testing on prod and am still having a hard time borrowing and reading books sometimes. It's so annoying because at this point I can't tell if the Axis360 errors are the problem or if it's something weird with our infrastructure. If these issues don't get resolved by Monday, I will probably tap in @Toxiapo to work with Greg and Keith to try to find the root of the problem. I have yet to see these issues on our Vercel infrastructure at all so it does seem like something might have changed recently that broke things for us.

**Comment by Toxiapo** [comment]
@samanthaandrews I think the only request that was never mocked was the request to validate the license in the axisnow package, I can't remember why on top of my head but I think I was having some problem intercepting that

**Comment by kristojorg** [comment]
Thank you @samanthaandrews! It's probably worth revisiting when we come back to our cypress tests in the future. One of the consequences of the monorepo is that flaky tests are going to slow a lot more people down, so I think in the future if we can't figure out how to make the cypress tests work consistently without ever reaching out to a server, we should switch back to more jest tests and maybe just a few integration or e2e tests for the most critical user flows

**Pull request merged**